mlir/utils/widgets/rock_conv.sh

```diff
@@ -0,0 +1,435 @@
+configVarToCL[filterLayout]=fil_layout
+configVarToCL[inputLayout]=in_layout
+configVarToCL[outputLayout]=out_layout
+configVarToCL[groupSize]=groupsize # can also use g
+configVarToCL[batchSize]=batchsize
+configVarToCL[inputChannel]=in_channels
+configVarToCL[outputChannel]=out_channels
+configVarToCL[inputHeight]=in_h
+configVarToCL[inputWidth]=in_w
+configVarToCL[filterHeight]=fil_h
+configVarToCL[filterWidth]=fil_w
+configVarToCL[dilationHeight]=dilation_h
+configVarToCL[dilationWidth]=dilation_w
+configVarToCL[strideHeight]=conv_stride_h
+configVarToCL[strideWidth]=conv_stride_w
+## padding_h
+configVarToCL[paddingHeightLeft]=padding_h_l
+configVarToCL[paddingHeightRight]=padding_h_r
+## padding_w
+configVarToCL[paddingWidthLeft]=padding_w_l
+configVarToCL[paddingWidthRight]=padding_w_r
+configVarToCL[tensorDataType]=t
+
+## Convert command line option to variable
+configCLToVar[operation]=operation
+configCLToVar[fil_layout]=filterLayout
+configCLToVar[in_layout]=inputLayout
+configCLToVar[out_layout]=outputLayout
+configCLToVar[groupsize]=groupSize
+configCLToVar[g]=groupSize
+configCLToVar[batchsize]=batchSize
+configCLToVar[in_channels]=inputChannel
+configCLToVar[out_channels]=outputChannel
+configCLToVar[in_h]=inputHeight
+configCLToVar[in_w]=inputWidth
+configCLToVar[fil_h]=filterHeight
+configCLToVar[fil_w]=filterWidth
+configCLToVar[dilation_h]=dilationHeight
+configCLToVar[dilation_w]=dilationWidth
+configCLToVar[conv_stride_h]=strideHeight
+configCLToVar[conv_stride_w]=strideWidth
+configCLToVar[padding_h_l]=paddingHeightLeft
+configCLToVar[padding_h_r]=paddingHeightRight
+configCLToVar[padding_w_l]=paddingWidthLeft
+configCLToVar[padding_w_r]=paddingWidthRight
+configCLToVar[t]=tensorDataType
+
+
+usage() {
+    cat <<END_USAGE
+Usage: $PROG [options...]
+
+Options:
+    -h|--help                  Display this help message
+    --input=                   Input test file.
+                               Note: input files can be either mlir test files
+                               (.mlir) or toml configuration files
+    --database=                Config database file(s). If in search mode, multiple
+                               database files can be selected if comma separated, e.g.
+                               $PROG --input=myTest --database=db1,db2 --search
+                               If in insert mode, only one database file is allowed.
+                               Note:
+                               1. database files must be .toml files.
+                               2. if database file is located at /mlir/test/e2e,
+                                  then only the filename without extension is needed.
+                                  Otherwiese, full file path is required.
+                               3. If no database files are provided, all toml files
+                                  at /mlir/test/e2e will be used.
+    --config                   A single config (pure + accessories) used to be
```

Comment:
I've been always confused about what is a config. The command line inputs to `rocmlir-gen` to generate a conv kernel includes parameters that
1. describe the geometry of the problem, like c, h, w, b, g of input, filter, and output
2. direction of the conv, like fwd, bwd, and wrw
3. computer related parameters, like data types, random inputs, and arch

Here I mean the parameters in the first category (geometry parameters) define a pure config. And other parameters are accessories to a pure config. This is used to determine if two configs are the same, i.e. we only compare the pure part of the config.
I was also thinking the parameters in the first two categories (geometry + direction) define a config mathematically, i.e. everything we need to compute a conv problem by hand. But I do not find this definition useful.

Reply:
Nice categorization, this is the first time I heard about the 3 categories and it make sense. But description here, `pure` + `accessories` seems to be misleading even after I understand about your definitions: It seems to imply `pure` or `geometry` side of configs must be complete, whereas you can pick and combine arbitrary number of `accessories` on top of `pure`. Is that the case? If so, you'll need to leave the definition inline here.

In reality my understanding is that your script will do the search regardless? So even if a user search for a config that has only the direction field (fwd), then the script will return all the `fwd` configs? If the script does not differentiate between the two concept then we should not put those here because it will be confusing to a user of the script.

Reply:
It is indeed misleading. Thank you for pointing that out.
The configs in the user input (command line or a file) can contain any parameters. Let's call them user-configs. The script processes the parameters in the user-config in the following three 3 scenarios:
1. Respect the parameter setting. This is the case for parameters in the geometry category
2. Ignore the parameter setting. This is the case for parameters in the direction and computer related categories
3. For parameters in the geometry category that are not set in the user-config, use the default value defined in `rocmlir-gen.cpp`.

In this way, each user-config can be converted into a "canonicalized " format.
The same process is done for configs in the target *.toml files. As a result, the search operation is done as a string match.

I feel like I need to include the above in the description. But it might be too long and verbose. I'll try to add a short summary in the description.